tomviz/LoadDataReaction.cxx

```diff
@@ -48,27 +50,40 @@ LoadDataReaction::~LoadDataReaction()
 //-----------------------------------------------------------------------------
 void LoadDataReaction::onTriggered()
 {
-  vtkNew<vtkSMParaViewPipelineController> controller;
+  loadData();
+}
 
-  // Let ParaView deal with reading the data. If we need more customization, we
-  // can show our own "File/Open" dialog and then create the appropriate reader
-  // proxies.
-  QList<pqPipelineSource*> readers = pqLoadDataReaction::loadData();
+QList<DataSource*> LoadDataReaction::loadData()
+{
+  vtkNew<vtkSMParaViewPipelineController> controller;
 
-  // Since we want to ditch the reader pipeline and just keep the raw data
-  // around. We do this magic.
-  foreach (pqPipelineSource* reader, readers)
+  QStringList filters;
+  filters << "Common file types (*.jpg *.jpeg *.png *.tiff *.tif *.raw *.dat *.bin *.txt)"
+          << "JPeg Image files (*.jpg *.jpeg)"
+          << "PNG Image files (*.png)"
+          << "TIFF Image files (*.tiff *.tif)"
+          << "Raw data files (*.raw)"
```

Comment:
The data and binary should all be lumped with Raw data files I think, as that is the reader we have to read them. We probably also need to do a little work to coerce that reader into being used.